nx/lib/nx/lin_alg.ex

```diff
@@ -949,6 +949,44 @@ defmodule Nx.LinAlg do
     )
   end
 
+  @doc """
+  Calculates the Moore-Penrose inverse, or the pseudoinverse, of a matrix.
+
+  ## Options
+    * `:eps` - Rounding error threshold used to assume values as 0.
+
+  ## Examples
+
+      iex> pinv = Nx.LinAlg.pinv(Nx.tensor([[1, 1], [3, 4]]))
+      #Nx.Tensor<
+        f32[2][2]
+        [
+          [4.0, -1.0000001192092896],
+          [-3.0, 1.0000001192092896]
+        ]
+      >
+
+      iex> Nx.LinAlg.pinv(Nx.tensor([[1, 3], [5, 4], [4, 9]]))
+      #Nx.Tensor<
+        f32[2][3]
+        [
+          [-0.07312049716711044, 0.3027806580066681, -0.11019568890333176],
+          [0.06900102645158768, -0.1307930052280426, 0.1462409943342209]
+        ]
+      >
```

Comment:
```suggestion
      iex> Nx.LinAlg.pinv(Nx.tensor(([[0.5, 0], [0, 1], [0.5, 0]]))
      #Nx.Tensor<
        f32[2][3]
        [
          [1.0, 0.0, 1.0],
          [0.0, 1.0, 0.0]
        ]
      >
```

There might be a rounding error or 2, but in general this example might look cleaner for a wide matrix